pkg/ccl/backupccl/backup_processor.go

```diff
@@ -118,17 +118,17 @@ func (bp *backupDataProcessor) Start(ctx context.Context) {
 
 // Next is part of the RowSource interface.
 func (bp *backupDataProcessor) Next() (rowenc.EncDatumRow, *execinfrapb.ProducerMetadata) {
-	if bp.State != execinfra.StateRunning {
-		return nil, bp.DrainHelper()
-	}
-
 	for prog := range bp.progCh {
 		// Take a copy so that we can send the progress address to the output
 		// processor.
 		p := prog
 		return nil, &execinfrapb.ProducerMetadata{BulkProcessorProgress: &p}
 	}
 
+	if bp.State != execinfra.StateRunning {
```

Comment:
I'm curious what motivated moving this down?

Reply:
@pbardea added an explanation to the commit message:
```
This test uncovered a situation in which a backup data processor
could be moved to state other than StatusRunning, which caused it
to exit early from the Next() method without draining the progCh. The goroutines responsible
for exporting the spans get stuck attempting to push to the progCh
that is only drained after the above StatusRunning check. This
change drains the progCh at the beginning of the Next() method.
```

Does this make sense? It seemed to mitigate the leaked goroutine in the backup test but wanted to run this by you.

Reply:
Got it, nice find. I think a lot of the bulk processors use this pattern so it may also need updating elsewhere.